dix/req.c

```diff
@@ -2270,11 +2317,21 @@ static int dix_cas_rops_alloc(struct m0_dix_req *req)
 		cas_rop->crp_cur_key = 0;
 	} m0_tl_endfor;
 
+	for (i = 0; i < skipped_sdevs_max && dtx != NULL && rc == 0; i++) {
+		if (skipped_sdevs[i] != NULL) {
+			cas_svc = pc->pc_dev2svc[*skipped_sdevs[i]].pds_ctx;
+			M0_ASSERT(cas_svc->sc_type == M0_CST_CAS);
+			M0_ASSERT(pa_idx < pa_nr);
+			rc = m0_dtx0_fid_assign(dtx, pa_idx,
+						&cas_svc->sc_fid);
+		}
+	}
 end:
 	if (rc != 0) {
 		dix_cas_rops_fini(&rop->dg_cas_reqs);
 		return M0_ERR(rc);
```

Comment:
as already pointed out in https://github.com/Seagate/cortx-motr/pull/1545, this might be a mem leak in error path.

Reply:
Yes and that fix will come to main as a part of merging [https://github.com/Seagate/cortx-motr/pull/1545](url)

Reply:
Hi @huanghua78 

https://github.com/Seagate/cortx-motr/pull/1545 is merged.  I think it does address your concern?

Reply:
Yes.